flink-clients/src/main/java/org/apache/flink/client/cli/GenericCLI.java

```diff
@@ -109,27 +104,29 @@ public void addGeneralOptions(Options baseOptions) {
 	}
 
 	@Override
-	public Configuration applyCommandLineOptionsToConfiguration(final CommandLine commandLine) {
-		final Configuration effectiveConfiguration = new Configuration(baseConfiguration);
+	public Configuration toConfiguration(final CommandLine commandLine) {
+		final Configuration resultConfiguration = new Configuration();
+
```

Comment:
Do we need this `if() {...}`? Eventually we will merge the configuration passed at the constructor to the one returned here, right? In addition, getting rid of accessing the outside configuration from here is one of the main points of this work, right?

Reply:
I see your point. For now we still use the `Configuration` in `isActive()`. We could think about not keeping the `Configuration` in the CLI but instead pass it as a parameter there as well. Also, I'm forwarding the option for now to make the returned `Configuration` "self contained", i.e. to have everything in there that the executor would expect.

What do you think?